qiskit/transpiler/passes/optimization/split_2q_unitaries.py

```diff
@@ -0,0 +1,76 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Splits each two-qubit gate in the `dag` into two single-qubit gates, if possible without error."""
+from typing import Optional
+
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.circuit.quantumcircuitdata import CircuitInstruction
+from qiskit.dagcircuit.dagcircuit import DAGCircuit, DAGOpNode
+from qiskit.circuit.library.generalized_gates import UnitaryGate
+from qiskit.synthesis.two_qubit.two_qubit_decompose import TwoQubitWeylDecomposition
+
+
+class Split2QUnitaries(TransformationPass):
+    """Splits each two-qubit gate in the `dag` into two single-qubit gates, if possible without error."""
```

Comment:
I feel like expanding the docs here to be a bit more explicit about what is going on in the pass would be good. But we can do this in a follow up for 1.2.0 final. If I were to read this I would be kind of confused by this docstring because I wouldn't understand how to split a 2q gate into two 1q gates without error.

```suggestion
    """Attempt to splits two-qubit gates in a :class:`.DAGCircuit` into two single-qubit gates
    
    This pass will analyze all the two qubit gates in the circuit and analyze the gate's unitary
    matrix to determine if the gate is actually a product of 2 single qubit gates. In these
    cases the 2q gate can be simplified into two single qubit gates and this pass will
    perform this optimization and will replace the two qubit gate with two single qubit
    :class:`.UnitaryGate`\s.
```